domains/client/block-builder/src/lib.rs

```diff
@@ -216,7 +212,21 @@ where
                     )),
                     Err(e) => TransactionOutcome::Rollback(Err(Error::from(e))),
                 }
-            })?;
+            });
+
+            match res {
+                Err(Error::ApplyExtrinsicFailed(ApplyExtrinsicFailed::Validity(e)))
+                    if e.exhausted_resources() =>
+                {
+                    tracing::debug!(
+                        "Reached block weight limit, skipping the remaining extrinsics at index {index}, total: {}",
+                        self.extrinsics.len()
+                    );
+                    return Ok(());
```

Comment:
This doesn't seem okay to me.
Resources exhaustion means that block was produced incorrectly and must be rejected, doesn't it?

Reply:
We don't actually execute the extrinsic while pushing them to the bundle, and the final domain block is unknown until the corresponding primary block is available, the actual weight consumed can't be known beforehand, it's inappropriate to say a domain block is invalid due to this reason.

Resources exhaustion simply means no more extrinsic can be executed, it's okay to have a simple return. As long as every executor processed the same list of extrinsics, their state is the same, otherwise, the fraud-proof should start to work.

Reply:
Then we just need to bump the limits so that they can't be exceeded, like `u64::MAX` for weight or whatnot.

BTW, do we have weight/size limits for individual bundles right now? I can't recall.

Reply:
Well, then bundle production is where limit should be placed. And then another limit for number of bundles per primary block depending on hardware expectations. But once bundles are produced and included in the primary chain, execution limits no longer make sense IMHO.

Having limits, but ignoring them just feels wrong.

Reply:
> Say a primary chain block includes the bundles exceeding the limit due to some bug, how it can be recovered if we choose to reject the domain block?

Will not happen unless we have a limit here, which I'm advocating to remove :upside_down_face: 

My point is that you either have a limit and return an error if it is exceeded, or you don't have a limit. What you seem to do is to have a limit, but ignore it printing debug message if it happens. That just doesn't make sense to me.

Reply:
Okay, I see the point, but having a limit is more appropriate since we do have an assumption on the executor hardware expectation IMHO. The reason to ignore it later is that practically makes more sense. Anyway, will let Jeremiah be the icebreaker.

Reply:
I'm not against having limits, I'm against having limits during execution. We should enforce limits during bundling and primary block production. This does change the rules of the game somewhat. We still need to track resources and stop execution if a particular extrinsic doesn't have enough fees to pay for it, but global limit no longer applies.

Reply:
>We should enforce limits during bundling and primary block production.

I agree with it, but that does not mean we can always make it, again, what if a program bug leads to the limit somehow not being able to be enforced? What I meant is apart from enforcing limits during bundling and primary block production(TODO), do not reject the domain block if it happens unfortunately.

Reply:
Then executors will have to deal with it and run whatever computation primary chain allowed to go through.

Reply:
Hmm, this's how executors handle that edge case, i.e., skip the remaining extrinsics and keep these already executed once resources are exhausted, no?

Reply:
what is needed to move this PR forward @nazar-pc @liuchengxu ? Is there still disagreement here, it appears from the last couple of comments you guys maybe agree...

Reply:
> Hmm, this's how executors handle that edge case, i.e., skip the remaining extrinsics and keep these already executed once resources are exhausted, no?

I feel like there is some communication issue here. Resources much be defined BEFORE executor is doing execution. It would be wrong and unexpected to execute bundle partially. We can abort execution of the particular extrinsic if it happens to be a contract and exceeds pre-paid resources, but otherwise there should be no restrictions here.